pkg/ccl/backupccl/manifest_handling.go

```diff
@@ -639,6 +639,16 @@ func FindPriorBackups(
 	return prev, nil
 }
 
+// IsCollectionRoot checks whether the directory pointed by store contains the latestFileName
+//pointer directory, indicating that the directory is the root of a backup collection
+func IsCollectionRoot(ctx context.Context, store cloud.ExternalStorage) (bool, error) {
+	_, err := store.ReadFile(ctx, latestFileName)
+	if errors.Is(err, cloud.ErrFileDoesNotExist) {
```

Comment:
We might need something along the lines of:

```
if err != nil {
			if errors.Is(err, cloud.ErrFileDoesNotExist) {
                ...
			}
			return false, pgerror.WithCandidateCode(err, pgcode.Io)
		}
```

this ensures that if we hit any other error apart from a file not exists, it gets propogated up the stack.

Reply:
done